src/entity/Guild/GuildUser.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  Entity,
+  Column,
+  OneToOne,
+  ManyToOne,
+  OneToMany,
+  PrimaryGeneratedColumn
+} from 'typeorm'
+import { Guild } from '.'
+import { GuildUserWarning } from './GuildUserWarning'
+import { GuildUserKick } from './GuildUserKick'
+import { GuildUserBan } from './GuildUserBan'
+import { GuildUserReputation } from './GuildUserReputation'
+import { User } from '../User'
+
+@Entity()
+export class GuildUser {
+  @PrimaryGeneratedColumn() id: number
+
+  @Column('varchar', { length: 100 })
+  nickname: string
+
+  @Column('timestamp without time zone') dateJoined: Date
+
+  @Column('timestamp without time zone', { nullable: true })
+  dateLastMessage: Date | null
+
+  @OneToMany(
+    type => GuildUserWarning,
+    guildUserWarning => guildUserWarning.user
+  )
+  warnings: GuildUserWarning[]
+
+  @OneToMany(type => GuildUserKick, guildUserKick => guildUserKick.user)
+  kicks: GuildUserKick[]
+
+  @OneToMany(type => GuildUserBan, guildUserBan => guildUserBan.user)
+  bans: GuildUserBan[]
+
+  @OneToOne(
+    type => GuildUserReputation,
+    guildUserReputation => guildUserReputation.user
+  )
+  reputation: GuildUserReputation
```

Comment:
Move the reputation to be on the User table. I want the user's reputation to follow them everywhere.